astrowidgets/interface_definition.py

```diff
@@ -0,0 +1,111 @@
+from typing import Protocol, runtime_checkable, Any
+from abc import abstractmethod
+
+# Allowed locations for cursor display
+ALLOWED_CURSOR_LOCATIONS = ['top', 'bottom', None]
+
+# List of marker names that are for internal use only
+RESERVED_MARKER_SET_NAMES = ['all']
+
+__all__ = [
+    'ImageViewerInterface',
+    'ALLOWED_CURSOR_LOCATIONS',
+    'RESERVED_MARKER_SET_NAMES'
+]
+
+
+@runtime_checkable
+class ImageViewerInterface(Protocol):
+    # This are attributes, not methods. The type annotations are there
+    # to make sure Protocol knows they are attributes. Python does not
+    # do any checking at all of these types.
+    click_center: bool
+    click_drag: bool
+    scroll_pan: bool
+    image_width: int
+    image_height: int
+    zoom_level: float
+    marker: Any
+    cuts: Any
+    stretch: str
+    # viewer: Any
+
+    # Allowed locations for cursor display
+    ALLOWED_CURSOR_LOCATIONS = ('top', 'bottom', None)
+
+    # List of marker names that are for internal use only
+    RESERVED_MARKER_SET_NAMES = ('all', )
+
+    # The methods, grouped loosely by purpose
+
+    # Methods for loading data
+    @abstractmethod
+    def load_fits(self, file):
```

Comment:
I know you say API change is out of scope, but I want to note that we agreed at Astropy Coordination Meeting 2023 to change `load_*` methods to just `load` so Astrowidgets does not dictate what file format each backend choose to support.

https://docs.google.com/document/d/1sBImbw1p4mM3VYWGQJDRbj2N3uqP5Ed_Wfb8BDjJj2o/edit#heading=h.8ah2ugluz5xr

Reply:
Thanks -- I added this to the agenda in the running notes.

Reply:
I opened https://github.com/astropy/astrowidgets/issues/163